lib/jekyll/theme.rb

```diff
@@ -0,0 +1,40 @@
+module Jekyll
+  class Theme
+    extend Forwardable
+    attr_reader :name
+    def_delegator :gemspec, :version, :version
+
+    def initialize(name)
+      @name = name.downcase.strip
+    end
+
+    def root
+      @root ||= gemspec.full_gem_path
+    end
+
+    def includes_path
+      path_for :includes
+    end
+
+    def layouts_path
+      path_for :layouts
+    end
+
+    def sass_path
+      path_for :sass
+    end
+
+    private
+
+    def path_for(folder)
+      path = Jekyll.sanitized_path root, "_#{folder}"
+      path if Dir.exists?(path)
+    end
+
+    def gemspec
+      @gemspec ||= Gem::Specification.find_by_name(name)
```

Comment:
@benbalter @parkr I love abusing enumerators and Gem::Specification provides a few:

```ruby
theme = Gem::Specification.find_all do |g|
  (g.name == name || \
   g.name =~ /\Ajekyll\-theme\-#{Regexp.escape(name)}\Z/)) && \
   g.files.grep(/\A_theme\//) 
end
```

Expand it how you see fit to enforce any kind of basic assurance a theme is a theme.